pkg/config/spec.go

```diff
@@ -722,3 +698,27 @@ func isMounted(config *Config, part *v1.Partition) (bool, error) {
 	}
 	return !notMnt, nil
 }
+
+func setSourceSize(cfg *Config, spec *v1.UpgradeSpec) error {
+	var size int64
+	var err error
+
+	if spec.RecoveryUpgrade {
+		size, err = GetSourceSize(cfg, spec.Recovery.Source)
+	} else {
+		size, err = GetSourceSize(cfg, spec.Active.Source)
+	}
+	if err != nil {
+		return err
+	}
+
+	cfg.Logger.Infof("Setting image size to %dMb", size)
+	// On upgrade only the active or recovery will be upgraded, so we dont need to override passive
+	if spec.RecoveryUpgrade {
```

Comment:
and here we do it again, so maybe it would be better if it's just done once at the beginning and then either spec.Recovery or spec.Active is handleded for the rest of the function

Reply:
Code is now refactored, this no longer applies.